src/main/java/io/jenkins/plugins/analysis/core/steps/IssuesPublisher.java

```diff
@@ -48,12 +48,14 @@
     private final QualityGateEvaluationMode qualityGateEvaluationMode;
     private final JobResultEvaluationMode jobResultEvaluationMode;
     private final LogHandler logger;
+    private final IssuesRecorder issuesRecorder;
 
     @SuppressWarnings("ParameterNumber")
     IssuesPublisher(final Run<?, ?> run, final AnnotatedReport report,
             final HealthDescriptor healthDescriptor, final QualityGateEvaluator qualityGate,
             final String name, final String referenceJobName, final boolean ignoreQualityGate,
-            final boolean ignoreFailedBuilds, final Charset sourceCodeEncoding, final LogHandler logger) {
+            final boolean ignoreFailedBuilds, final Charset sourceCodeEncoding, final LogHandler logger
+    ,final IssuesRecorder issuesRecorder) {
```

Comment:
Rather than using parameter `issueRecorder` you need to provide the boolean `failOnErrors `.

Reply:
But is that value not set in `IssuesRecorder` and can be obtained from its getter ?